aws/resource_aws_cur_report_definition.go

```diff
@@ -0,0 +1,189 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/costandusagereportservice"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"log"
+)
+
+func resourceAwsCurReportDefinition() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCurReportDefinitionCreate,
+		Read:   resourceAwsCurReportDefinitionRead,
+		Delete: resourceAwsCurReportDefinitionDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"report_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 256),
+			},
+			"time_unit": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					costandusagereportservice.TimeUnitDaily,
+					costandusagereportservice.TimeUnitHourly,
+				}, false),
+			},
+			"format": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					costandusagereportservice.ReportFormatTextOrcsv}, false),
+			},
+			"compression": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					costandusagereportservice.CompressionFormatGzip,
+					costandusagereportservice.CompressionFormatZip,
+				}, false),
+			},
+			"additional_schema_elements": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						costandusagereportservice.SchemaElementResources,
+					}, false),
+				},
+				Set:      schema.HashString,
+				Required: true,
+				ForceNew: true,
+			},
+			"s3_bucket": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"s3_prefix": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(0, 256),
+			},
+			"s3_region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"additional_artifacts": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						costandusagereportservice.AdditionalArtifactQuicksight,
+						costandusagereportservice.AdditionalArtifactRedshift,
+					}, false),
+				},
+				Set:      schema.HashString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCurReportDefinitionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).costandusagereportconn
+
+	reportName := *aws.String(d.Get("report_name").(string))
```

Comment:
This is a little confusing to look at, I'd recommend:

```suggestion
	reportName := d.Get("report_name").(string)
```

And creating the pointer only in the one spot its necessary below:

```go
		ReportName:  aws.String(reportName),
```